### DISCUSSION TITLE
Save coordinates and avoid recalculate all data

### DISCUSSION THREAD

**Discussion by croonos23** [networkx/networkx#4859] [Q&A]
This is my code, I add a node but I want to keep the position (x, y) of the first iteration and only calculate the position of the new element

import networkx as nx
import matplotlib.pyplot as plt

G = nx.Graph()

G.add_edges_from([
("P","1"),
("P","2"),
("P","3"),
("P","4"),
("P","5"),
("P","6"),
("P","7")

])
pos = nx.planar_layout(G, scale=1000)
nx.draw_networkx_nodes(G,pos)
nx.draw_networkx_edges(G,pos,edgelist=G.edges(),edge_color="black")
nx.draw_networkx_labels(G,pos)
print(pos)
plt.show()

{'P': array([-1000.        ,  -333.33333333]), '1': array([ 523.80952381, -333.33333333]), '2': array([-238.0952381 ,  428.57142857]), '3': array([-111.11111111,  301.58730159]), '4': array([ 15.87301587, 174.6031746 ]), '5': array([142.85714286,  47.61904762]), '6': array([269.84126984, -79.36507937]), '7': array([ 396.82539683, -206.34920635])}

**Comment by dschult** [comment]
The `planar_layout` function does not provide an input argument that fixes a subset of the nodes while requesting positions for the rest.  Indeed, that may not be easy to implement in general. The problem is different from laying out all the nodes. Other layout algorithms do provide this capability, but don't provide a planar layout.
